controller/payload/payload_scheduler.py

```diff
@@ -60,6 +60,8 @@
 lf_exec_env_image = os.getenv("LF_EXEC_ENV_IMAGE")
 ml_exec_env_image = os.getenv("ML_EXEC_ENV_IMAGE")
 exec_env_network = os.getenv("LF_NETWORK")
+if config_service.get_config_value("is_managed"):
```

Comment:
as far as i know we usually don't run conditions etc in the .py file directly. 
Since the config service also needs to be present for this to work (which it always should but who know :D ) I would go maybe with a slightly different approach. Singleton...ish:

```
inference_dir = "NOT_INITIALZED"

def get_finerence_dir():
   if inference_dir == "NOT_INITIALZED"
       if config_service.get_config_value("is_managed"):
           inference_dir = os.getenv("INFERENCE_DIR")
       else:
           inference_dir = None
   return inference_dir
```

and wherever we need the value we can use the function. This would then only collect the info if/when necessary when possible other startup issues are already handled.

Reply:
This works, if we assume that the config for "is_managed" does not change during runtime. Is that fine or should we also handle this edge case?

Reply:
changing at runtime isn't supposed to happen atm (all our guides to change include a restart of gateway & config service).
That said we could also go without the global variable and just use the function since the overhead for getting the config value & os variable should be negligible:
```python
# inference_dir = "NOT_INITIALZED"

def get_finerence_dir() -> str:
   if config_service.get_config_value("is_managed"):
       return os.getenv("INFERENCE_DIR")
   return None

```

Reply:
changed